### DISCUSSION TITLE
Entities best practices

### DISCUSSION THREAD

**Discussion by terrypascal** [abpframework/abp#8765] [General]
On the following page: https://docs.abp.io/en/abp/latest/Best-Practices/Entities

The suggestion is to validate a property, that an independent "setter" method should be used, and then to make private the property setter.  

C#.NET was developed with this concept in mind so that validation could be added at any time during an application life cycle without affecting class users/clients.

private string _title;
public virtual string Title 
{
    get => _title;
    set => _title = Check.NotNullOrWhiteSpace(value, nameof(value));
}

There are already built in getters and setters.  To be used for validation, simply write the validator logic within the setter.

**Comment by enisn** [comment]
If the database has somehow data with empty title, your select query which includes that data will throw always that exception. 

Even ORM can't set that property while reading from database.

**Reply by terrypascal** [reply]
So you think it's a good thing that you have validation on a field, but that the database should be able to have data that doesn't pass validation?

So the user could select data from the database, change a different field, click "update" and the validation would fail??

What is the purpose of the property get and set methods then?
